Error when running first notebook cell block of "Querying for data" section

Got the following error when running cell block https://aiida-tutorials.readthedocs.io/en/tutorial-2022-intro/sections/managing_data/querying.html#querying-for-data from notebook. I guess this might be an aiida-core issue? @chrisjsewell 

```python
09/27/2022 10:08:53 AM <2081> sqlalchemy.pool.impl.QueuePool: [ERROR] Error closing cursor
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/cursor.py", line 1099, in fetchmany
    new = dbapi_cursor.fetchmany(size - lb)
psycopg2.ProgrammingError: named cursor isn't valid anymore

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1995, in _safe_close_cursor
    cursor.close()
psycopg2.ProgrammingError: named cursor isn't valid anymore
---------------------------------------------------------------------------
ProgrammingError                          Traceback (most recent call last)
/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/cursor.py in fetchmany(self, result, dbapi_cursor, size)
   1098             try:
-> 1099                 new = dbapi_cursor.fetchmany(size - lb)
   1100             except BaseException as e:

ProgrammingError: named cursor isn't valid anymore

The above exception was the direct cause of the following exception:

ProgrammingError                          Traceback (most recent call last)
<ipython-input-1-916cde5a3649> in <module>
    134         structure.set_extra('formula', structure.get_formula(mode='count'))
    135 
--> 136 store_formula_in_extra()

<ipython-input-1-916cde5a3649> in store_formula_in_extra()
    131     query = QueryBuilder()
    132     query.append(StructureData, filters={'extras':{'!has_key':'formula'}})
--> 133     for structure, in query.iterall():
    134         structure.set_extra('formula', structure.get_formula(mode='count'))
    135 

/opt/conda/lib/python3.9/site-packages/aiida/orm/querybuilder.py in iterall(self, batch_size)
   1048         :returns: a generator of lists
   1049         """
-> 1050         for item in self._impl.iterall(self.as_dict(), batch_size):
   1051             # Convert to AiiDA frontend entities (if they are such)
   1052             for i, item_entry in enumerate(item):

/opt/conda/lib/python3.9/site-packages/aiida/storage/psql_dos/orm/querybuilder/main.py in iterall(self, data, batch_size)
    167             stmt = build.query.statement.execution_options(yield_per=batch_size)
    168 
--> 169             for resultrow in self.get_session().execute(stmt):
    170                 yield [self.to_backend(rowitem) for rowitem in resultrow]
    171 

/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/result.py in iterrows(self)
    380 
    381             def iterrows(self):
--> 382                 for row in self._fetchiter_impl():
    383                     row = make_row(row) if make_row else row
    384                     if post_creational_filter:

/opt/conda/lib/python3.9/site-packages/sqlalchemy/orm/loading.py in chunks(size)
    140 
    141             if yield_per:
--> 142                 fetch = cursor.fetchmany(yield_per)
    143 
    144                 if not fetch:

/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/result.py in fetchmany(self, size)
   1095         """
   1096 
-> 1097         return self._manyrow_getter(self, size)
   1098 
   1099     def all(self):

/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/result.py in manyrows(self, num)
    540                     num = real_result._yield_per
    541 
--> 542                 rows = self._fetchmany_impl(num)
    543                 if make_row:
    544                     rows = [make_row(row) for row in rows]

/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/cursor.py in _fetchmany_impl(self, size)
   1806 
   1807     def _fetchmany_impl(self, size=None):
-> 1808         return self.cursor_strategy.fetchmany(self, self.cursor, size)
   1809 
   1810     def _raw_row_iterator(self):

/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/cursor.py in fetchmany(self, result, dbapi_cursor, size)
   1099                 new = dbapi_cursor.fetchmany(size - lb)
   1100             except BaseException as e:
-> 1101                 self.handle_exception(result, dbapi_cursor, e)
   1102             else:
   1103                 if not new:

/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/cursor.py in handle_exception(self, result, dbapi_cursor, err)
    939 
    940     def handle_exception(self, result, dbapi_cursor, err):
--> 941         result.connection._handle_dbapi_exception(
    942             err, None, None, dbapi_cursor, result.context
    943         )

/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/base.py in _handle_dbapi_exception(self, e, statement, parameters, cursor, context)
   2122                 util.raise_(newraise, with_traceback=exc_info[2], from_=e)
   2123             elif should_wrap:
-> 2124                 util.raise_(
   2125                     sqlalchemy_exception, with_traceback=exc_info[2], from_=e
   2126                 )

/opt/conda/lib/python3.9/site-packages/sqlalchemy/util/compat.py in raise_(***failed resolving arguments***)
    206 
    207         try:
--> 208             raise exception
    209         finally:
    210             # credit to

/opt/conda/lib/python3.9/site-packages/sqlalchemy/engine/cursor.py in fetchmany(self, result, dbapi_cursor, size)
   1097         if size > lb:
   1098             try:
-> 1099                 new = dbapi_cursor.fetchmany(size - lb)
   1100             except BaseException as e:
   1101                 self.handle_exception(result, dbapi_cursor, e)

ProgrammingError: (psycopg2.ProgrammingError) named cursor isn't valid anymore
(Background on this error at: https://sqlalche.me/e/14/f405)
```

--- a/docs/sections/managing_data/groups.md
+++ b/docs/sections/managing_data/groups.md
@@ -156,7 +156,7 @@ Success: Group<my_group> deleted.
 
 Any deletion operation related to groups will not affect the nodes themselves.
 For example if you delete a group, the nodes that belonged to the group will remain in the database.
-The same happens if you remove nodes from the group -- they will remain in the database but won't belong to the group any more. 
+The same happens if you remove nodes from the group -- they will remain in the database but won't belong to the group any more.
 You can delete all nodes in the group along with the group itself by adding option `--delete-nodes` to the `verdi group delete` command.
 
 :::

--- a/docs/sections/managing_data/groups.md
+++ b/docs/sections/managing_data/groups.md
@@ -104,7 +104,7 @@ $ verdi group relabel a_group my_group
 Success: Label changed to my_group
 ```
 
-Add one or more nodes to your new group using node PKs from the `tutorial_pbesol` group we inspected earlier (the PK might be different from your profile):
+Add one or more nodes to your new group using node PKs from the `tutorial_pbesol` group we inspected earlier (the PK might be different for your database):
 
 ```{code-block} console
 $ verdi group add-nodes -G my_group 380 1273

--- a/docs/sections/managing_data/groups.md
+++ b/docs/sections/managing_data/groups.md
@@ -107,7 +107,7 @@ Success: Label changed to my_group
 Add one or more nodes to your new group using node PKs from the `tutorial_pbesol` group we inspected earlier (the PK might be different for your database):
 
 ```{code-block} console
-$ verdi group add-nodes -G my_group 380 1273
+$ verdi group add-nodes -G my_group <PK> <PK>
 Do you really want to add 2 nodes to Group<my_group>? [y/N]: y
 ```
 
@@ -140,7 +140,7 @@ PK    Type         Created
 Removing nodes from the group is similar to adding them, try:
 
 ```{code-block} console
-$ verdi group remove-nodes -G my_group 380
+$ verdi group remove-nodes -G my_group <PK>
 Do you really want to remove 1 nodes from Group<my_group>? [y/N]: y
 ```
 

--- a/docs/sections/managing_data/groups.md
+++ b/docs/sections/managing_data/groups.md
@@ -104,7 +104,7 @@ $ verdi group relabel a_group my_group
 Success: Label changed to my_group
 ```
 
-Add one or more nodes to your new group using node PKs from the `tutorial_pbesol` group we inspected earlier:
+Add one or more nodes to your new group using node PKs from the `tutorial_pbesol` group we inspected earlier (the PK might be different from your profile):
 
 ```{code-block} console
 $ verdi group add-nodes -G my_group 380 1273
@@ -132,15 +132,15 @@ Group description  <no description>
 # Nodes:
 PK    Type         Created
 ----  -----------  -----------------
-74  CalcJobNode  2078D:17h:51m ago
-76  CalcJobNode  2078D:17h:57m ago
+ 380  CalcJobNode  2078D:17h:46m ago
+1273  CalcJobNode  2078D:18h:03m ago
 ...
 ```
 
 Removing nodes from the group is similar to adding them, try:
 
 ```{code-block} console
-$ verdi group remove-nodes -G my_group <PK>
+$ verdi group remove-nodes -G my_group 380
 Do you really want to remove 1 nodes from Group<my_group>? [y/N]: y
 ```
 
@@ -156,7 +156,8 @@ Success: Group<my_group> deleted.
 
 Any deletion operation related to groups will not affect the nodes themselves.
 For example if you delete a group, the nodes that belonged to the group will remain in the database.
-The same happens if you remove nodes from the group -- they will remain in the database but won't belong to the group any more.
+The same happens if you remove nodes from the group -- they will remain in the database but won't belong to the group any more. 
+You can delete all nodes in the group along with the group itself by adding option `--delete-nodes` to the `verdi group delete` command.
 
 :::
 

--- a/docs/sections/managing_data/querying.ipynb
+++ b/docs/sections/managing_data/querying.ipynb
@@ -177,7 +177,8 @@
     "    from aiida.orm import QueryBuilder\n",
     "    query = QueryBuilder()\n",
     "    query.append(StructureData, filters={'extras':{'!has_key':'formula'}})\n",
-    "    for structure, in query.iterall():\n",
+    "    structures = query.all(flat=True)\n",
+    "    for structure in structures:\n",
     "        structure.set_extra('formula', structure.get_formula(mode='count'))\n",
     "\n",
     "store_formula_in_extra()"

--- a/docs/sections/managing_data/querying.ipynb
+++ b/docs/sections/managing_data/querying.ipynb
@@ -1949,7 +1949,7 @@
     "\n",
     "Alternatively, the dataset is included in the virtual machine image of the AiiDA tutorial from [aiida-tutorials.readthedocs.io](https://aiida-tutorials.readthedocs.io) (choose the correct version of the tutorial, depending on which version of AiiDA you want to try).\n",
     "\n",
-    "If the data is not available yet, you can use the following command to import the required calculations (if you are in a notebook you can execute the command in cell by add an exclamation mark `!` in front)::\n",
+    "If the data is not available yet, you can use the following command to import the required calculations (if you are in a notebook you can execute the command in a cell by adding an exclamation mark `!` in front)::\n",
     "\n",
     "```{code-block} console\n",
     "verdi archive import https://object.cscs.ch/v1/AUTH_b1d80408b3d340db9f03d373bbde5c1e/marvel-vms/tutorials/aiida_tutorial_2022_10_perovskites_main_0001.aiida\n",

--- a/docs/sections/managing_data/querying.ipynb
+++ b/docs/sections/managing_data/querying.ipynb
@@ -1774,8 +1774,8 @@
     "It will return a dictionary with all the attributes the node has.\n",
     "\n",
     "Note that a node also has a number of additional methods and attributes.\n",
-    "For instance, you can do `node.attributes_keys()` to get only the attribute keys or `node.get_attribute('wfc_cutoff')` to get the value of a single attribute (these two variants are more efficient if the node has a lot of attributes and you don't need all data).\n",
-    "Similarly, for `extras`, you have `node.extras`, `node.extras_keys()`, and `node.get_attribute('SOME_EXTRA_KEY')`."
+    "For instance, you can do `node.base.attributes.keys()` to get only the attribute keys or `node.base.attributes.get('wfc_cutoff')` to get the value of a single attribute (these two variants are more efficient if the node has a lot of attributes and you don't need all data).\n",
+    "Similarly, for `extras`, you have `node.base.extras.all`, `node.base.extras.keys()`, and `node.base.attributes.get('<SOME_EXTRA_KEY>')`."
    ]
   },
   {
@@ -1802,8 +1802,8 @@
     "query = QueryBuilder()\n",
     "query.append(PwCalculation)\n",
     "node, = query.first()\n",
-    "print('Attributes dictionary:', node.attributes, '\\n')\n",
-    "print('Extras dictionary:', node.extras)"
+    "print('Attributes dictionary:', node.base.attributes.all, '\\n')\n",
+    "print('Extras dictionary:', node.base.extras.all)"
    ]
   },
   {

--- a/docs/sections/managing_data/querying.ipynb
+++ b/docs/sections/managing_data/querying.ipynb
@@ -1949,7 +1949,7 @@
     "\n",
     "Alternatively, the dataset is included in the virtual machine image of the AiiDA tutorial from [aiida-tutorials.readthedocs.io](https://aiida-tutorials.readthedocs.io) (choose the correct version of the tutorial, depending on which version of AiiDA you want to try).\n",
     "\n",
-    "If the data is not available yet, you can use the following command to import the required calculations:\n",
+    "If the data is not available yet, you can use the following command to import the required calculations (if you are in a notebook you can execute the command in cell by add an exclamation mark `!` in front)::\n",
     "\n",
     "```{code-block} console\n",
     "verdi archive import https://object.cscs.ch/v1/AUTH_b1d80408b3d340db9f03d373bbde5c1e/marvel-vms/tutorials/aiida_tutorial_2022_10_perovskites_main_0001.aiida\n",
@@ -1969,7 +1969,7 @@
     "These calculations are spin-polarized (without spin-orbit coupling), use a Gaussian smearing and perform a variable-cell relaxation of the full unit cell.\n",
     "The goal of this part of the tutorial is to have you utilize what you have learnt in the previous sections and “screen” for magnetic and metallic perovskites in a “high-throughput” way.\n",
     "As you learned previously in the tutorial, AiiDA allows to organize calculations into groups.\n",
-    "Once more check the list of groups in your database by typing:"
+    "Once more check the list of groups in your database by typing (From jupyter notebook or `IPython`):"
    ]
   },
   {
@@ -2282,14 +2282,6 @@
    "source": [
     "plot_results(results)"
    ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "461a8d21",
-   "metadata": {},
-   "outputs": [],
-   "source": []
   }
  ],
  "metadata": {

--- a/docs/sections/managing_data/querying.ipynb
+++ b/docs/sections/managing_data/querying.ipynb
@@ -177,9 +177,8 @@
     "    from aiida.orm import QueryBuilder\n",
     "    query = QueryBuilder()\n",
     "    query.append(StructureData, filters={'extras':{'!has_key':'formula'}})\n",
-    "    structures = query.all(flat=True)\n",
-    "    for structure in structures:\n",
-    "        structure.set_extra('formula', structure.get_formula(mode='count'))\n",
+    "    for structure, in query.all():\n",
+    "        structure.base.extras.set('formula', structure.get_formula(mode='count'))\n",
     "\n",
     "store_formula_in_extra()"
    ]

